storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1374,5 +1393,29 @@ public static RuntimeException wrapInRuntime(Exception e){
             return new RuntimeException(e);
         }
     }
+
+    public static int getAvailablePort(Integer port) {
+        Integer localPort = null;
+        ServerSocket socket = null;
+        try {
+            socket = new ServerSocket(port);
+            localPort = socket.getLocalPort();
+        } catch (IOException exp) {
+            getAvailablePort(0);
+        } finally {
```

Comment:
Can we use the java 7 closable to make this code a little simpler?  And why is port an Integer and not an int?

```
public static int getAvailablePort(int port) throws IOException {
    try (ServerSocket socket = new ServerSocket(port)) {
        return socket.getLocalPort();
    } catch (IOException e) {
        if (port <= 0) {
            throw e;
        }
        return getAvailablePort(0);
    }
}
```